sql/core/src/main/scala/org/apache/spark/sql/execution/joins/BroadcastHashJoinExec.scala

```diff
@@ -453,6 +494,47 @@ case class BroadcastHashJoinExec(
     val (keyEv, anyNull) = genStreamSideJoinKey(ctx, input)
     val (matched, checkCondition, _) = getJoinCondition(ctx, input)
     val numOutput = metricTerm(ctx, "numOutputRows")
+    val isLongHashedRelation = broadcastRelation.value.isInstanceOf[LongHashedRelation]
+
+    // fast stop if isOriginInputEmpty = true
+    // whether isNullAwareAntiJoin is true or false
+    // should accept all rows in streamedSide
+    if (broadcastRelation.value.isOriginInputEmpty) {
+      return s"""
+                |// Common Anti Join isOriginInputEmpty(true) accept all
+                |$numOutput.add(1);
+                |${consume(ctx, input)}
+          """.stripMargin
+    }
+
+    if (isNullAwareAntiJoin) {
+      if (broadcastRelation.value.allNullColumnKeyExistsInOriginInput) {
+        return s"""
+                  |// NAAJ isOriginInputEmpty(false) anyNullKeyExists(true) reject all
```

Comment:
Can you paste the gencode in this condition ? 

It appears that we would run an empty loop here: 
```
while (stream side has rows) {
// drop the row
}
```

Should we instead like omit the whole loop ? Or will that be too complex ? How is an "Always false" filter handled in spark ?

Reply:
the for loop is generated in parent codegen support operator. 
if we adopt "${consume(ctx, input)}" means this row is adopted and placed into downstream operator.
if we want to reject all column it's as simple as just putting a empty string in BHJ scope gen code